### ISSUE TITLE
gql-gen should be saved either as a dev dependency or globally

### ISSUE THREAD

**Issue by jeffreyyoung** [dotansimha/graphql-code-generator#1410]
**Describe the bug**
The docs say that the module should be installed with the following command: `npm install graphql-code-generator`.   I think it should be installed in either of the following ways: 
* as a dev dependency, `npm install -D graphql-code-generator`
* or globally `npm install -g graphql-code-generator`

I thiiiiiiiink since the code generator is only run in development environments, it doesn't make sense to install it as a dependency.

**To Reproduce**
1. Read the README.md and install the way it tells you to install gql-gen as a project dependency

**Expected behavior**
gql-gen should be installed as a dev dependency or globally

**Additional context**
I could be totally wrong.

**Comment by dotansimha** [comment]
Hi @jeffreyyoung !
You are right, our best-practice is to install the codegen locally in the project, as dev-dependency, and to avoid installing it as global package.

**Comment by dotansimha** [comment]
Fix available in 1.0.0 👍

**Comment by usersina** [comment]
Running the app with only production dependencies `npm install --production` will give the following error:

**generated/graphql.ts**:
```JAVASCRIPT
import gql from 'graphql-tag';
                 ^^^^^^^^^^^^
Cannot find module 'graphql-tag' or its corresponding type declarations.
```


Installing the `graphql-tag` package separately will make `node main.js` exit on entry without any output with exit code 1
